internal/http/listener.go

```diff
@@ -136,20 +137,39 @@ func newHTTPListener(ctx context.Context, serverAddrs []string) (listener *httpL
 		}
 	}()
 
+	// Silently ignore failure to bind on DNS cached ipv6 loopback
 	for _, serverAddr := range serverAddrs {
 		var l net.Listener
 		if l, err = listenCfg.Listen(ctx, "tcp", serverAddr); err != nil {
+			if strings.HasPrefix(serverAddr, "[::1]") {
```

Comment:
Maybe let's log instead of doing this strings check in L144 and L153, then return when `len(tcpListeners) == 0`. as you already did below. This looks easier.

Also won't IPv4 be dead anytime soon ?

Reply:
Please take another look. I've removed the explicit ipv6 loopback string check. Is this what you meant?

Reply:
Hi @vadmeste I need to revert this change since we want to see the failure if any of the explicit ipv4 addresses provided is invalid. We don't want to silently ignore this. Let me know your thoughts please.

Also, I spoke to @dvaldivia today and added more changes. We're want to gracefully let minio startup if the user **explicitly** tries to bind using ```localhost``` and the failure is on the ipv6 loopback.